app/src/org/odk/collect/android/widgets/EthiopianDateWidget.java

```diff
@@ -1,532 +1,118 @@
-         * Update the widget helper date text (useful for those who don't know the Ethiopian calendar)
-         * @param dtGreg
-         */
-        private void updateGregorianDateHelperDisplay(){
-            DateTime dtLMDGreg = getCurrentEthiopianDateDisplay().withChronology(GregorianChronology.getInstance());
-            DateTimeFormatter fmt = DateTimeFormat.forPattern("d MMMM yyyy");
-            String str = fmt.print(dtLMDGreg);
-            txtGregorian.setText("("+str+")");
-        }
-       
-        /**
-         * Listens for button being pressed by touchscreen
-         * @author alex
-         */
-        private class EDWTouchListener implements OnTouchListener{
-            private View mView;
-            private Handler mHandler;
-            public EDWTouchListener(View mV, Handler mH){
-                mView = mV;
-                mHandler = mH;
-            }
-            /*
-             * (non-Javadoc)
-             * @see android.view.View.OnTouchListener#onTouch(android.view.View, android.view.MotionEvent)
-             */
-            @Override
-            public boolean onTouch(View v, MotionEvent event) {
-                boolean isReleased = event.getAction() == MotionEvent.ACTION_UP || event.getAction() == MotionEvent.ACTION_CANCEL;
-                boolean isPressed = event.getAction() == MotionEvent.ACTION_DOWN;
-
-                if (isReleased) {
-                    stopUpdating();
-                } else if (isPressed) {
-                    startUpdating(v == mView,mHandler);
-                }
-                return false;
-            }
-        }
-        
-        /**
-         * Listens for button being pressed by keypad/trackball
-         * @author alex
-         */
-        private class EDWKeyListener implements OnKeyListener{
-            private View mView;
-            private Handler mHandler;
-            public EDWKeyListener(View mV, Handler mH){
-                mView = mV;
-                mHandler = mH;
-            }
-            /*
-             * (non-Javadoc)
-             * @see android.view.View.OnKeyListener#onKey(android.view.View, int, android.view.KeyEvent)
-             */
-            @Override
-            public boolean onKey(View v, int keyCode, KeyEvent event) {
-                boolean isKeyOfInterest = keyCode == KeyEvent.KEYCODE_DPAD_CENTER || keyCode == KeyEvent.KEYCODE_ENTER;
-                boolean isReleased = event.getAction() == KeyEvent.ACTION_UP;
-                boolean isPressed = event.getAction() == KeyEvent.ACTION_DOWN
-                        && event.getAction() != KeyEvent.ACTION_MULTIPLE;
-
-                if (isKeyOfInterest && isReleased) {
-                    stopUpdating();
-                } else if (isKeyOfInterest && isPressed) {
-                    startUpdating(v == mView,mHandler);
-                }
-                return false;
-            }
-        }
+public class EthiopianDateWidget extends AbstractUniversalDateWidget {
```

Comment:
There is no Chronology implementation for the Nepali calendar, so the way each handles date manipulation is quite different.